### PR TITLE
Add ToC to readme

### PR DESCRIPTION
**Pull request by adamdecaf**
I heard about this on [Go Time](https://changelog.com/gotime/66) and want to help out. Here's a Table of Contents in the readme. 

You can view it better [on my fork](https://github.com/adamdecaf/go-perfbook/tree/readme-ToC).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# go-prefbook
```
Comment (dgryski) [inline README.md]: Typo

**Comment by dgryski** [comment]
LGTM. Thanks for doing this. I think once the book is a bit more fleshed out we'll be able to move to an auto-generated ToC.

File: README.md (hunk 1 above)
Comment (adamdecaf) [reply README.md]: Fixed.

**Comment by adamdecaf** [comment]
Sounds great!

**Pull request merged**